### DISCUSSION TITLE
Allow customization for visual appearance of @storybook/addon-themes addon

### DISCUSSION THREAD

**Discussion by soosap** [storybookjs/storybook#24988] [Ideas]
### Is your feature request related to a problem? Please describe.

![CleanShot 2023-11-26 at 22 15 51@2x](https://github.com/storybookjs/storybook/assets/7187705/16bc7f7c-c2e1-4daa-82e4-a52118ca462a)

The visual appearance of this addon is not that nice. All controls in the control panel take up the same amount of space. But for some reason the dark/light toggle can only be rendered with 5x the real estate that is given to other controls.

### Describe the solution you'd like

It would be nice if we could configure the visual appearance. I'd prefer something more in the lines of:

![CleanShot 2023-11-26 at 22 17 47@2x](https://github.com/storybookjs/storybook/assets/7187705/0131659f-f69f-413d-a31a-05100a098f35)

In other words, customization should allow to just use:
* "icon-only"
* "icon" + "mode name" i.e. `dark`, `light`. 
* "icon" + "mode name" + "suffix" i.e. `dark theme`, `light theme` OR `dark mode`, `light mode` 

Also the terms `mode` and `theme` are used interchangeably. That is adding a lot of confusion to my project, in which I refer to dark/light as **mode** and to `primary`, `secondary`, `danger`, etc. as **theme**.  The control hard coding the terms "x-theme" and "y-theme", while `theme` means something else in my code base is a pain.

### Are you able to assist to bring the feature to reality?

no
